lib/config.js

```diff
@@ -216,8 +216,56 @@ function parseConfig(config) {
   )
 }
 
+/**
+ * Decodes a base64 encoded string into a config
+ *
+ * @param {string} encodedRuleSet A base64 encoded string that needs decoding
+ * @param {array} [processed] List of config files already processed, used to prevent loops
+ * @returns {Object} The loaded repolinter json config
+ * @throws Will throw an error if unable to parse config or if config is invalid
+ */
+async function decodeConfig(encodedRuleSet, processed = []) {
+  const configData = Buffer.from(encodedRuleSet, 'base64').toString()
+
+  let ruleset
+  // try parsing as JSON, then YAML
+  try {
+    ruleset = JSON.parse(configData)
+  } catch (je) {
+    try {
+      ruleset = yaml.safeLoad(configData)
+    } catch (ye) {
+      throw new Error(
+        `unable to parse ruleset as either JSON (error: ${je}) or YAML (error: ${ye})`
+      )
+    }
+  }
+
+  // merge extended rulesets
+  // TODO: Verify functionality when used in conjunction with encoded rulesets... no config location available
```

Comment:
This is something that was being discussed in the slack: how do you specify custom resources (extended rulesets or custom rules) when the ruleset is being served from another location. I'm not sure we landed on a decision but when we do we can address this TODO.

Reply:
I was struggling with this as well. Unfortunately, I have not found a solution yet. I decided that I would park it for a fix later, as we did not need these extended rulesets YET. 
We have several discussions going on within Philips businesses to add the ability to extend rulesets, so may need to start working on it once we figure out a way to move forward.  Should we open a separate GH Discussion/Issue for this?

Reply:
+1 to digging into this discussion more. My assumption on custom resources, especially custom rules, was a PATH-like envrionment variable pointing to a directory; however that's probably me feeling too 'local'.

Reply:
You can now pass on base64 encoded string in 'extend' section.